### PR TITLE
test: Refactor smoke tests and add improve nightly workflow

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

This PR refactors smoke tests to be more useable and reliable
- Smoke tests can be defined in each resource/datasource test files with function prefix `TestSmokeTests_`
- Slack notification will be sent at the end of the workflow to notify the slack channel with build result and url
e.g. 

## ✔️ How to Test

Forked Run - https://github.com/ykim-1/terraform-provider-linode/actions/runs/10603716867

`make smoke-test`

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: .github/workflows/nightly_smoke_tests.yml
Comment (ykim-1) [reply .github/workflows/nightly_smoke_tests.yml:46]: It is possible but I found this one to be more easy to use and implement

Comment (ykim-1) [reply .github/workflows/nightly_smoke_tests.yml:45]: I tried to use both versions but getting the right payload for the official version is way harder. I don't think using the rtCamp/action-slack-notify will cause any issues and it is widely used by others. I will change to the official version if there is any problem in future

File: linode/acceptance/util.go
Comment (ykim-1) [inline linode/acceptance/util.go:693]: This map variable contains all the smoke test cases across test packages in each directory. We will need to add new smoke tests to this map

File: linode/childaccount/datasource_test.go
Comment (ykim-1) [inline linode/childaccount/datasource_test.go:13]: we will need to add new smoke tests in this function for each test file to allow `make smoke-test` pick up the new tests

File: .github/workflows/nightly_smoke_tests.yml
Comment (lgarber-akamai) [reply .github/workflows/nightly_smoke_tests.yml:45]: Any chance we could pin this action to a specific commit hash in that case?

Comment (ykim-1) [reply .github/workflows/nightly_smoke_tests.yml:45]: Yes, looks like commit hash value is included in the slack message. 
e.g.
image

File: linode/childaccount/datasource_test.go
Comment (zliang-akamai) [inline linode/childaccount/datasource_test.go:25]: So can't we just include all smoke tests in the `tests` variable here without the global `SmokeTests`?

Comment (ykim-1) [reply linode/childaccount/datasource_test.go:25]: That's correct. I had to do this because `func TestSmokeTests_...` need to be defined in every test file that contains smoke test(s) and I wanted one global place where we can see the entire suite easily.

Comment (ykim-1) [reply linode/childaccount/datasource_test.go:25]: That is true, I will remove the global map for now then

File: .github/workflows/nightly_smoke_tests.yml
Comment (ykim-1) [reply .github/workflows/nightly_smoke_tests.yml]: Ended up using the official version because Slack Webhooks are going to be deprecated soon

File: Makefile
Comment (jriddle-linode) [reply Makefile:85]: This is great, needed in all our repos that use e2e lol.

**Review by jriddle-linode — APPROVED** [review]
Looks good can't wait.

File: .github/workflows/nightly_smoke_tests.yml
Comment (zliang-akamai) [inline .github/workflows/nightly_smoke_tests.yml:45]:
```suggestion
        uses: slackapi/slack-github-action@v1.27.0
```
The 1.27 version of the action is available now

**Review by zliang-akamai — APPROVED** [review]
Looks great, local run passed, nice work!

**Review by lgarber-akamai — APPROVED** [review]
All tests are passing and the workflow changes look good, nice work!

**Comment by ykim-1** [comment]
Still waiting on slack admin approval to post the notification

**Pull request merged**
